### PR TITLE
[soxr] Add new port

### PR DESCRIPTION
**Pull request by ThijsWithaar**
Request to add the [soxr](https://sourceforge.net/p/soxr/wiki/Home/) library, see also #3557.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @ThijsWithaar, thanks for the new port. Here are the test results from the current CI system:

|x64-osx|master| test |notes
|-------|------|------|-----
|soxr   |      |Fail  |New Port

|arm64-windows|master| test |notes
|-------------|------|------|-----
|soxr         |      |Fail  |New Port

|x64-linux|master| test |notes
|---------|------|------|-----
|soxr     |      |Fail  |New Port

|x64-uwp|master| test |notes
|-------|------|------|-----
|soxr   |      |Fail  |New Port

|arm-uwp|master| test |notes
|-------|------|------|-----
|soxr   |      |Fail  |New Port

[failureLogs.zip](https://github.com/microsoft/vcpkg/files/3189490/failureLogs.zip)

File: ports/soxr/disable_cmake_settings.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -40,10 +40,10 @@
+     "Build type, one of: None Debug Release RelWithDebInfo MinSizeRel." FORCE)
+ endif ()
+ 
+-option (BUILD_TESTS "Build sanity-tests." ON)
```
Comment (Voskrese) [inline ports/soxr/disable_cmake_settings.patch]: also easier through the port file

vcpkg_configure_cmake(
    SOURCE_PATH ${SOURCE_PATH}
    PREFER_NINJA
    OPTIONS
        -DBUILD_TESTS=OFF
        -DWITH_OPENMP=OFF
        -DWITH_LSR_BINDINGS=OFF

....

**Review by MVoz — COMMENTED** [review]
hmm...

**Comment by ThijsWithaar** [comment]
Ok, update to use cmake settings instead of a path. It seems I still have a warning:
```
CMake Warning at C:/vsts/_work/7/s/scripts/buildsystems/vcpkg.cmake:107 (message):
  There are no libraries installed for the Vcpkg triplet arm64-windows.
```
And an error which needs addressing:
```
vector-gen.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.
```
It does build fine on my machine, so I'm not completely sure what's going on.

Is it obvious for you what's going on ?

**Comment by grdowns** [comment]
Thanks for the new port, @ThijsWithaar!

**Pull request merged**